src/app/components/todos/todos.component.ts

```diff
@@ -20,14 +20,14 @@ export class TodosComponent implements OnInit {
   }
 
   deleteTodo(todo:Todo){
-    this.todos = this.todos.filter(t=>t.id!==todo.id);
-    this.todoService.deleteTodo(todo).subscribe();
+    this.todoService.addTodo(todo).subscribe(todo => {
+      this.todos.push(todo);
+    );
   }
 
```

Comment:
In the deleteTodo method, the logic seems to be incorrect. It should be removing the todo from the list, not adding it. Please review the logic and ensure it is deleting the correct todo.